frontend/src/app/shared/components/suggested-challenge/suggested-challenge.component.ts

```diff
@@ -1,47 +1,103 @@
+import { ChallengePreview } from '@shared/models/challenge/challenge-preview';
+import { SuggestionSettings } from '@shared/models/challenge/suggestion-settings';
+import { Language } from '@shared/models/language/language';
+import { takeUntil } from 'rxjs';
 
 @Component({
     selector: 'app-suggested-challenge',
     templateUrl: './suggested-challenge.component.html',
     styleUrls: ['./suggested-challenge.component.sass'],
 })
-export class SuggestedChallengeComponent {
-    @Input() tags: string[] = ['Algorithms', 'Strings', 'Data Types', 'Formatting', 'Logic', 'Graph', 'Stack'];
-
-    @Input() constraints: string[] = [
-        'Strings passed in will consist of only letters and spaces.',
-        'Spaces will be included only when more than one word is present.',
-        'Some another constraint.',
-        'Some another constraint.',
-        'Some another constraint.',
-    ];
+export class SuggestedChallengeComponent extends BaseComponent implements OnInit {
+    challenge?: ChallengePreview;
 
-    @Input() languages: string[] = [
-        'C#',
-        'Javascript',
-    ];
+    languagesNames: string[] = [];
+
+    languages: Language[] = [];
+
+    suggestionTypesNames: string[];
 
-    score: string[] = [
-        'Fundamentals',
-        'Rank up',
-        'Practice and Repeat',
-        'Beta',
-        'Random',
+    suggestionTypes = [
+        { name: 'Fundamentals', id: 0 },
+        { name: 'Rank up', id: 1 },
+        { name: 'Practice and Repeat', id: 2 },
+        { name: 'Beta', id: 3 },
+        { name: 'Random', id: 4 },
     ];
 
-    @Input() description: string = 'Write a function that takes in a string of one or more words,'
-    + 'and returns the same string, but with all five or more letter, words reversed.';
+    suggestionSettings: SuggestionSettings = {
+        languageId: 0,
+        suggestionType: 0,
+    };
+
+    constructor(
+        private challengeService: ChallengeService,
+        private languageService: LanguageService,
+        private toastrService: ToastrNotificationsService,
+    ) { super(); }
+
+    public ngOnInit(): void {
+        this.getLanguages();
+
+        this.suggestionTypesNames = this.suggestionTypes.map(item => item.name);
+    }
 
-    @Input() title: string = 'Stop gninnipS My sdroW!';
+    public onLanguageChanged(value: string | string[]) {
+        if (typeof value !== 'string') {
+            return;
+        }
+        this.suggestionSettings.languageId = this.languages.find(item => item.name === value)?.id ?? 0;
+        this.getChallenge();
+    }
 
-    @Input() difficulty: string = '6 kyu';
+    public onSuggestionTypeChanged(value: string | string[]) {
+        if (typeof value !== 'string') {
+            return;
+        }
+        this.suggestionSettings.suggestionType = this.suggestionTypes.find(item => item.name === value)?.id ?? 0;
+        this.getChallenge();
+    }
```

Comment:
54-55 and 62-64 almost the same
move all this logic to one generic method and use it

Reply:
moved to function findItemIdByName(collection: SimpleCollection[], name: string) in suggested-challenge.utils.ts